sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityGenerator.scala

```diff
@@ -36,9 +36,9 @@ class UnivocityGenerator(
   writerSettings.setHeaders(schema.fieldNames: _*)
   private val gen = new CsvWriter(writer, writerSettings)
 
-  // A `ValueConverter` is responsible for converting a value of an `InternalRow` to `String`.
+  // A `ValueConverter` is responsible for converting a value of an `Any` to `String`.
   // When the value is null, this converter should not be called.
-  private type ValueConverter = (InternalRow, Int) => String
```

Comment:
`(InternalRow, Int)` is more efficient than `Any`, as we can avoid boxing for primitive values.

Reply:
I get that we want to recursively call the converter for array, then we can use `(SpecializedGetters, Int)`

Reply:
Done.